Cargo.toml

```diff
@@ -418,7 +418,7 @@ tracing = { version = "0.1.40", features = ["std"] }
 tracing-appender = "0.2.3"
 tracing-opentelemetry = "0.23.0"
 tracing-span-tree = "0.1"
-tracing-subscriber = { version = "0.3.18", features = [
+tracing-subscriber = { version = "0.3.20", features = [
```

Comment:
Similarly here, maybe just have version as "0.3"?

Reply:
I don't mind, but for some reason we have patch version pinned here, so I'd rather keep it then change without understanding
@nagisa what do you think?

Reply:
Cargo interprets dependencies with no other qualifiers as

```
^version
```

which is a short for

```
>=version, <next_semver_break(version)
```

---

This means that even though `0.3.18` is specified here, cargo can lock any version in between `0.3.18` to `0.4` (which is the next semver break.) For versions above 0 major only major version changes are considered breaking.

---

Specifying 0.3.20 here is actually the correct use of dependency specification. We do not want cargo to pick up one of the vulnerable versions ever again.